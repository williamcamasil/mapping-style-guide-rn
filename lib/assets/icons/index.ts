export * as Large from './large';
export * as Heavy from './heavy';
export * as Thin from './thin';
export * as Small from './small';
export * as Default from './default';
