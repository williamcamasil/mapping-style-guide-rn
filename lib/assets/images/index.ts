export { default as BtnTakePicture } from './BtnTakePicture';
export { default as BulletPoint } from './BulletPoint';
export { default as CheckboxImg } from './CheckboxImg';
export { default as DialogClose } from './DialogClose/index.svg';
export { default as Lock } from './Lock';
export { default as MaskedCardNumber } from './MaskedCardNumber';
export { default as Pointer } from './Pointer';
export { default as RadioImg } from './RadioImg';
