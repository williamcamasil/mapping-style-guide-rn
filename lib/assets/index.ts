export * as Icons from './icons';
export * as Images from './images';
export * as Lotties from './lotties';
