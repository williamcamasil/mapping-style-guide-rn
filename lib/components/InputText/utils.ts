import { TextStyle, ViewStyle, Platform } from 'react-native';

import { AppThemeType } from '../../theme/default';
import { ColorPaletesType } from '../../tokens';
import { calcLineHeight } from '../../utils/typography';

const getBorder = (theme: AppThemeType, focused: boolean, disabled?: boolean, textArea?: boolean) => {
  if (textArea) return null;

  return {
    borderColor: focused ? theme.colors.neutralGray400 : theme.colors.neutralGray300,
    borderRadius: theme.borders.radius.medium,
    borderWidth: disabled ? theme.borders.width.none : theme.borders.width.thin,
  };
};

export const getViewStyle = (
  theme: AppThemeType,
  height: number,
  focused: boolean,
  hasValue: boolean,
  textArea?: boolean,
  disabled?: boolean,
): ViewStyle => ({
  height,
  backgroundColor: disabled ? theme.colors.neutralGray100 : theme.colors.neutralWhite,
  ...getBorder(theme, focused, disabled, textArea),
  paddingLeft: !textArea ? theme.spacings.sSmall : theme.spacings.sZero,
  paddingTop: Platform.select({
    android: hasValue ? height / 2 : undefined,
    ios: hasValue ? height / 2 : 4,
  }),
  paddingBottom: Platform.select({
    android: hasValue ? height / 8 : undefined,
    ios: hasValue ? height / 16 : 0,
  }),
});

const getTextColor = (focused?: boolean, disabled?: boolean): ColorPaletesType => {
  if (disabled) return 'neutralGray500';

  if (focused) return 'neutralGray700';

  return 'neutralGray600';
};

export const getTextStyle = (
  theme: AppThemeType,
  focused: boolean,
  hasValue: boolean,
  disabled?: boolean,
): TextStyle => {
  const fontSize = hasValue ? theme.typography.sizes.medium : theme.typography.sizes.small;
  return {
    ...(hasValue ? theme.typography.weights.bold : theme.typography.weights.regular),
    fontSize,
    letterSpacing: theme.typography.letterSpaces.default,
    lineHeight: calcLineHeight(fontSize, theme.typography.lineHeights.small),
    color: theme.colors[getTextColor(focused, disabled)],
  };
};

export const getErrorStyle = (theme: AppThemeType, error?: boolean): ViewStyle | null => {
  if (!error) return null;

  return {
    borderColor: theme.colors.feedbackError400,
  };
};

export const getActionStyle = (secureTextEntry?: boolean): ViewStyle | null => {
  if (!secureTextEntry) return null;

  return {
    paddingRight: 50,
  };
};

export const getTextLabelColor = (focused?: boolean, error?: boolean, textArea?: boolean): ColorPaletesType => {
  if (error && textArea) {
    return 'feedbackError400';
  }

  return focused ? 'neutralGray600' : 'neutralGray500';
};

export const INPUT_TEXT_HEIGHT = 64;
