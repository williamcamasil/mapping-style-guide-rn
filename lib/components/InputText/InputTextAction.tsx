import React, { ComponentType } from 'react';
import {
  ColorValue, StyleSheet, TouchableOpacity,
} from 'react-native';

import { SvgProps } from 'react-native-svg';

import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import { INPUT_TEXT_HEIGHT } from './utils';

export type InputTextActionPropType<IconProps extends SvgProps = SvgProps> = {
  theme: AppThemeType;
  onPress: () => void;
  Icon: ComponentType<IconProps>;
  color?: ColorValue;
  size: number;
  testID?: string;
  disabled?: boolean;
};

const styled = StyleSheet.create({
  icon: {
    position: 'absolute',
    justifyContent: 'center',
    alignItems: 'center',
    right: 0,
    top: 0,
    height: INPUT_TEXT_HEIGHT,
    width: INPUT_TEXT_HEIGHT,
  },
});

const InputTextAction: React.FC<InputTextActionPropType> = ({
  theme,
  onPress: handlePress,
  Icon,
  size,
  color,
  testID,
  disabled,
}) => (
  <TouchableOpacity
    testID={testID}
    style={styled.icon}
    onPress={handlePress}
    activeOpacity={theme.opacities.intense}
    disabled={disabled}
  >
    <Icon width={size} height={size} color={color || theme.colors.neutralGray400} />
  </TouchableOpacity>
);

InputTextAction.defaultProps = {
  size: 24,
  testID: 'input-text-action',
};

export default withTheme(InputTextAction);
