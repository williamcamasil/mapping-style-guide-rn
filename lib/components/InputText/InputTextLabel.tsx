import React from 'react';

import { useTextStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import { calcLineHeight } from '../../utils/typography';
import Text from '../Text';
import { getTextLabelColor } from './utils';

type InputTextLabelPropsType = {
  children?: string;
  hide?: boolean;
  focused?: boolean;
  error?: boolean;
  textArea?: boolean;
  theme: AppThemeType;
};

const InputTextLabel: React.FC<InputTextLabelPropsType> = ({
  children, hide, focused, error, textArea, theme,
}) => {
  const labelStyles = useTextStyles(() => [
    {
      position: 'absolute',
      left: !textArea ? theme.spacings.sSmall : theme.spacings.sZero,
      top: theme.spacings.sXXS,
      lineHeight: calcLineHeight(theme.typography.sizes.XXS, theme.typography.lineHeights.small),
      color: theme.colors[getTextLabelColor(focused, error, textArea)],
    },
  ], [
    theme.spacings,
    theme.typography.sizes.XXS,
    theme.typography.lineHeights.small,
    theme.colors,
    focused,
    error,
    textArea,
  ]);

  if (hide) return null;

  return (
    <Text variant="caption" color="neutralGray500" style={labelStyles}>
      {children}
    </Text>
  );
};

InputTextLabel.defaultProps = {
  hide: false,
};

export default withTheme(InputTextLabel);
