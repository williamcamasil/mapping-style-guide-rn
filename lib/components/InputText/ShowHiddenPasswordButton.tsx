import React, { useCallback } from 'react';

import { Icons } from '../../assets';
import InputTextAction from './InputTextAction';

type ShowHiddenPasswordButtonPropsType = {
  visible?: boolean;
  showPassword: boolean;
  disabled?: boolean;
  onShowPasswordChange: (value: boolean) => void;
};

const ShowHiddenPasswordButton: React.FC<ShowHiddenPasswordButtonPropsType> = ({
  visible,
  showPassword,
  disabled,
  onShowPasswordChange,
}) => {
  const handlePress = useCallback(() => {
    onShowPasswordChange(!showPassword);
  }, [onShowPasswordChange, showPassword]);

  if (!visible) return null;

  return (
    <InputTextAction
      testID="show-or-hidden-password-touchable"
      Icon={showPassword ? Icons.Default.EyeOff : Icons.Default.EyeOn}
      size={24}
      onPress={handlePress}
      disabled={Boolean(disabled)}
    />
  );
};

export default ShowHiddenPasswordButton;
