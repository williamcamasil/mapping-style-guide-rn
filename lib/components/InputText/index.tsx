import React, { useCallback, useState } from 'react';
import {
  NativeSyntheticEvent,
  StyleProp,
  TextInput as RNTextInput,
  TextInputFocusEventData,
  TextInputProps,
  View,
  ViewStyle,
} from 'react-native';

import { useTextStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import { getFieldErrorState, wrapField } from '../Form';
import HelpMessage from '../HelpMessage';
import InputTextAction, { InputTextActionPropType } from './InputTextAction';
import InputTextLabel from './InputTextLabel';
import ShowHiddenPasswordButton from './ShowHiddenPasswordButton';
import {
  getErrorStyle,
  getTextStyle,
  getViewStyle,
  getActionStyle,
  INPUT_TEXT_HEIGHT,
} from './utils';

export type InputTextPropsType = Omit<TextInputProps, 'placeholder' | 'onChange' | 'onChangeText' | 'editable' | 'pointerEvents'> & {
  theme: AppThemeType;
  label: string;
  containerStyle?: StyleProp<ViewStyle>;
  value: string | null;
  onChange?: TextInputProps['onChangeText'];
  help?: string;
  error?: boolean;
  disabled?: boolean;
  /**
   * Desabilita o Input sem alterar o estilo.
   */
  readOnly?: boolean;
  action?: React.ReactElement<InputTextActionPropType, typeof InputTextAction>;
  inputPointerEvents?: TextInputProps['pointerEvents'];
  innerRef?: React.LegacyRef<RNTextInput>;
  textArea?: boolean;
};

const InputText = ({
  theme,
  style,
  label,
  value,
  containerStyle,
  onChange,
  onFocus,
  onBlur,
  help,
  error,
  disabled,
  readOnly,
  secureTextEntry,
  action,
  inputPointerEvents,
  innerRef,
  textArea,
  ...others
}: InputTextPropsType): React.ReactElement => {
  const [focused, setFocused] = useState(false);

  const [showPassword, setShowPassword] = useState(false);

  const hasValue = Boolean(value);
  const hasAction = Boolean(action ?? secureTextEntry);

  const inputStyles = useTextStyles(
    () => [
      getTextStyle(theme, focused, hasValue, disabled),
      getViewStyle(theme, INPUT_TEXT_HEIGHT, focused, hasValue, textArea, disabled),
      getErrorStyle(theme, error),
      getActionStyle(hasAction),
      style,
    ],
    [theme, focused, hasValue, disabled, textArea, error, hasAction, style],
  );

  const handleFocus = useCallback(
    (e: NativeSyntheticEvent<TextInputFocusEventData>) => {
      setFocused(true);
      onFocus?.(e);
    },
    [onFocus],
  );

  const handleBlur = useCallback(
    (e: NativeSyntheticEvent<TextInputFocusEventData>) => {
      setFocused(false);
      onBlur?.(e);
    },
    [onBlur],
  );

  const handleShowOrHidden = useCallback((show: boolean) => {
    setShowPassword(show);
  }, []);

  const renderAction = () => {
    if (action) {
      return action;
    }

    return (
      <ShowHiddenPasswordButton
        visible={secureTextEntry}
        showPassword={showPassword}
        onShowPasswordChange={handleShowOrHidden}
        disabled={disabled}
      />
    );
  };

  return (
    <View style={containerStyle}>
      <RNTextInput
        testID="rn-text-input"
        autoCapitalize={secureTextEntry ? 'none' : undefined}
        autoComplete={secureTextEntry ? 'off' : undefined}
        autoCorrect={secureTextEntry ? false : undefined}
        spellCheck={secureTextEntry ? false : undefined}
        {...others}
        ref={innerRef}
        secureTextEntry={Boolean(secureTextEntry && !showPassword)}
        editable={Boolean(!disabled && !readOnly)}
        onFocus={handleFocus}
        onBlur={handleBlur}
        onChangeText={onChange}
        value={value}
        placeholderTextColor={theme.colors.neutralGray500}
        placeholder={label}
        style={inputStyles}
        pointerEvents={inputPointerEvents}
      />
      <HelpMessage
        error={error}
        disabled={disabled}
        marginLeft={!textArea ? theme.spacings.sSmall : theme.spacings.sZero}
      >
        {help}
      </HelpMessage>

      <InputTextLabel hide={!hasValue} focused={focused} error={error} textArea={textArea}>
        {label}
      </InputTextLabel>

      {renderAction()}
    </View>
  );
};

InputText.defaultProps = {
  allowFontScaling: false,
};

InputText.Field = withTheme(wrapField<InputTextPropsType>((props, fieldProps) => {
  const {
    invalid, touched, error, submitError, value, onChange, onBlur, onFocus,
  } = fieldProps;

  const { showError, message } = getFieldErrorState(
    invalid,
    touched,
    error,
    submitError,
    props.help,
  );

  return (
    <InputText
      {...props}
      value={value || ''}
      onChange={onChange}
      onBlur={onBlur}
      onFocus={onFocus}
      error={showError}
      help={message}
    />
  );
}));

InputText.Action = InputTextAction;

export default withTheme(InputText);
