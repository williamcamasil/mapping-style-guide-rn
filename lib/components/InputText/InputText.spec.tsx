import React from 'react';

import { within, render, fireEvent } from '@testing-library/react-native';

import InputText from '.';
import { Icons } from '../../assets';
import { DefaultTheme } from '../../theme';

describe('InputText snapshot', () => {
  it('empty', () => {
    const tree = render(
      <InputText
        label="Input label"
        value=""
      />,
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('default', () => {
    const tree = render(
      <InputText
        label="Input label"
        value="Input value"
        help="Help message"
      />,
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('disabled', () => {
    const tree = render(
      <InputText
        label="Input label"
        value="Input value"
        help="Help message"
        disabled
      />,
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('readOnly', () => {
    const tree = render(
      <InputText
        label="Input label"
        value="Input value"
        help="Help message"
        readOnly
      />,
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('error', () => {
    const tree = render(
      <InputText
        label="Input label"
        value="Input value"
        error
        help="Error message"
      />,
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('focused', () => {
    const result = render(
      <InputText
        label="Input label"
        value="Input value"
        error
        help="Error message"
      />,
    );

    fireEvent(result.getByTestId('rn-text-input'), 'focus');

    const tree = result.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('secureTextEntry', () => {
    const result = render(<InputText label="Input label" value="Input value" secureTextEntry />);

    const tree = result.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('action', () => {
    const result = render(<InputText
      label="Input label"
      value="Input value"
      action={(
        <InputText.Action
          Icon={Icons.Small.Down}
          size={16}
          color={DefaultTheme.colors.neutralGray500}
          onPress={jest.fn()}
        />
      )}
    />);

    const tree = result.toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('InputText events', () => {
  it('onFocus & onBlur', () => {
    const handleFocus = jest.fn();
    const handleBlur = jest.fn();

    const result = render(
      <InputText
        label="Input label"
        value=""
        onFocus={handleFocus}
        onBlur={handleBlur}
      />,
    );

    const nativeInput = result.getByTestId('rn-text-input');

    fireEvent(nativeInput, 'focus');

    expect(handleFocus).toBeCalled();

    fireEvent(nativeInput, 'blur');

    expect(handleBlur).toBeCalled();
  });

  it('onChange', () => {
    const handleChange = jest.fn();

    const result = render(
      <InputText
        label="Input label"
        value=""
        onChange={handleChange}
      />,
    );

    fireEvent(result.getByTestId('rn-text-input'), 'changeText', 'type text');

    expect(handleChange).toBeCalledWith('type text');
  });

  it('onChange disabled', () => {
    const handleChange = jest.fn();

    const result = render(
      <InputText
        label="Input label"
        value=""
        onChange={handleChange}
        disabled
      />,
    );

    fireEvent(result.getByTestId('rn-text-input'), 'changeText', 'type text');

    expect(handleChange).not.toBeCalled();
  });

  it('Should verify if password is active', () => {
    const result = render(<InputText label="Input label" value="" secureTextEntry />);

    const element = result.getByTestId('rn-text-input');

    expect(element.props.secureTextEntry).toBe(true);
  });

  it('Should click in button buttonShowOrHidden and disable secureTextEntry and should visible showOrHiddenPassword with eyeOff', () => {
    const result = render(<InputText label="Input label" value="" secureTextEntry />);

    const rnTextInput = result.getByTestId('rn-text-input');

    const buttonShowOrHidden = result.getByTestId('show-or-hidden-password-touchable');

    expect(rnTextInput.props.secureTextEntry).toBe(true);

    fireEvent.press(buttonShowOrHidden);

    const icon = within(buttonShowOrHidden).UNSAFE_getByType(Icons.Default.EyeOff);
    expect(icon).toBeTruthy();

    expect(rnTextInput.props.secureTextEntry).toBe(false);
  });

  it('Should click in button buttonShowOrHidden and disable secureTextEntry and after click in button again should visible showOrHiddenPassword with eyeOn', () => {
    const result = render(<InputText label="Input label" value="" secureTextEntry />);

    const rnTextInput = result.getByTestId('rn-text-input');

    const buttonShowOrHidden = result.getByTestId('show-or-hidden-password-touchable');

    expect(rnTextInput.props.secureTextEntry).toBe(true);

    fireEvent.press(buttonShowOrHidden);

    expect(rnTextInput.props.secureTextEntry).toBe(false);
    const iconOff = within(buttonShowOrHidden).UNSAFE_getByType(Icons.Default.EyeOff);
    expect(iconOff).toBeTruthy();

    fireEvent.press(buttonShowOrHidden);

    expect(rnTextInput.props.secureTextEntry).toBe(true);
    const iconOn = within(buttonShowOrHidden).UNSAFE_getByType(Icons.Default.EyeOn);
    expect(iconOn).toBeTruthy();
  });
});
