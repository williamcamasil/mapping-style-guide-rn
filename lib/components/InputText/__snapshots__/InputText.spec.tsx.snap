// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText snapshot action 1`] = `
<View>
  <TextInput
    allowFontScaling={false}
    editable={true}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Input label"
    placeholderTextColor="#A4A7AF"
    secureTextEntry={false}
    style={
      {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#E1E2E4",
        "borderRadius": 12,
        "borderWidth": 1,
        "color": "#676D79",
        "fontFamily": "Averta CY",
        "fontSize": 19,
        "fontWeight": "bold",
        "height": 64,
        "letterSpacing": -0.02,
        "lineHeight": 19,
        "paddingBottom": 4,
        "paddingLeft": 20,
        "paddingRight": 50,
        "paddingTop": 32,
      }
    }
    testID="rn-text-input"
    value="Input value"
  />
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#A4A7AF",
        "flexShrink": 1,
        "fontFamily": "Averta CY",
        "fontSize": 13,
        "fontWeight": "normal",
        "left": 20,
        "letterSpacing": -0.02,
        "lineHeight": 13,
        "position": "absolute",
        "top": 12,
      }
    }
  >
    Input label
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "height": 64,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 64,
      }
    }
    testID="input-text-action"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="16"
      bbWidth="16"
      fill="none"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      vbHeight={16}
      vbWidth={16}
      width={16}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="m4 6 4 4 4-4"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4288980911,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`InputText snapshot default 1`] = `
<View>
  <TextInput
    allowFontScaling={false}
    editable={true}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Input label"
    placeholderTextColor="#A4A7AF"
    secureTextEntry={false}
    style={
      {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#E1E2E4",
        "borderRadius": 12,
        "borderWidth": 1,
        "color": "#676D79",
        "fontFamily": "Averta CY",
        "fontSize": 19,
        "fontWeight": "bold",
        "height": 64,
        "letterSpacing": -0.02,
        "lineHeight": 19,
        "paddingBottom": 4,
        "paddingLeft": 20,
        "paddingTop": 32,
      }
    }
    testID="rn-text-input"
    value="Input value"
  />
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#676D79",
        "flexShrink": 1,
        "fontFamily": "Averta CY",
        "fontSize": 14,
        "fontWeight": "normal",
        "letterSpacing": -0.02,
        "lineHeight": 14,
        "marginLeft": 20,
        "marginTop": 12,
      }
    }
  >
    Help message
  </Text>
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#A4A7AF",
        "flexShrink": 1,
        "fontFamily": "Averta CY",
        "fontSize": 13,
        "fontWeight": "normal",
        "left": 20,
        "letterSpacing": -0.02,
        "lineHeight": 13,
        "position": "absolute",
        "top": 12,
      }
    }
  >
    Input label
  </Text>
</View>
`;

exports[`InputText snapshot disabled 1`] = `
<View>
  <TextInput
    allowFontScaling={false}
    editable={false}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Input label"
    placeholderTextColor="#A4A7AF"
    secureTextEntry={false}
    style={
      {
        "backgroundColor": "#F7F7F8",
        "borderColor": "#E1E2E4",
        "borderRadius": 12,
        "borderWidth": 0,
        "color": "#A4A7AF",
        "fontFamily": "Averta CY",
        "fontSize": 19,
        "fontWeight": "bold",
        "height": 64,
        "letterSpacing": -0.02,
        "lineHeight": 19,
        "paddingBottom": 4,
        "paddingLeft": 20,
        "paddingTop": 32,
      }
    }
    testID="rn-text-input"
    value="Input value"
  />
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#A4A7AF",
        "flexShrink": 1,
        "fontFamily": "Averta CY",
        "fontSize": 14,
        "fontWeight": "normal",
        "letterSpacing": -0.02,
        "lineHeight": 14,
        "marginLeft": 20,
        "marginTop": 12,
      }
    }
  >
    Help message
  </Text>
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#A4A7AF",
        "flexShrink": 1,
        "fontFamily": "Averta CY",
        "fontSize": 13,
        "fontWeight": "normal",
        "left": 20,
        "letterSpacing": -0.02,
        "lineHeight": 13,
        "position": "absolute",
        "top": 12,
      }
    }
  >
    Input label
  </Text>
</View>
`;

exports[`InputText snapshot empty 1`] = `
<View>
  <TextInput
    allowFontScaling={false}
    editable={true}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Input label"
    placeholderTextColor="#A4A7AF"
    secureTextEntry={false}
    style={
      {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#E1E2E4",
        "borderRadius": 12,
        "borderWidth": 1,
        "color": "#676D79",
        "fontFamily": "Averta CY",
        "fontSize": 16,
        "fontWeight": "normal",
        "height": 64,
        "letterSpacing": -0.02,
        "lineHeight": 16,
        "paddingBottom": 0,
        "paddingLeft": 20,
        "paddingTop": 4,
      }
    }
    testID="rn-text-input"
    value=""
  />
</View>
`;

exports[`InputText snapshot error 1`] = `
<View>
  <TextInput
    allowFontScaling={false}
    editable={true}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Input label"
    placeholderTextColor="#A4A7AF"
    secureTextEntry={false}
    style={
      {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#E53930",
        "borderRadius": 12,
        "borderWidth": 1,
        "color": "#676D79",
        "fontFamily": "Averta CY",
        "fontSize": 19,
        "fontWeight": "bold",
        "height": 64,
        "letterSpacing": -0.02,
        "lineHeight": 19,
        "paddingBottom": 4,
        "paddingLeft": 20,
        "paddingTop": 32,
      }
    }
    testID="rn-text-input"
    value="Input value"
  />
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#E53930",
        "flexShrink": 1,
        "fontFamily": "Averta CY",
        "fontSize": 14,
        "fontWeight": "normal",
        "letterSpacing": -0.02,
        "lineHeight": 14,
        "marginLeft": 20,
        "marginTop": 12,
      }
    }
  >
    Error message
  </Text>
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#A4A7AF",
        "flexShrink": 1,
        "fontFamily": "Averta CY",
        "fontSize": 13,
        "fontWeight": "normal",
        "left": 20,
        "letterSpacing": -0.02,
        "lineHeight": 13,
        "position": "absolute",
        "top": 12,
      }
    }
  >
    Input label
  </Text>
</View>
`;

exports[`InputText snapshot focused 1`] = `
<View>
  <TextInput
    allowFontScaling={false}
    editable={true}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Input label"
    placeholderTextColor="#A4A7AF"
    secureTextEntry={false}
    style={
      {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#E53930",
        "borderRadius": 12,
        "borderWidth": 1,
        "color": "#3E4149",
        "fontFamily": "Averta CY",
        "fontSize": 19,
        "fontWeight": "bold",
        "height": 64,
        "letterSpacing": -0.02,
        "lineHeight": 19,
        "paddingBottom": 4,
        "paddingLeft": 20,
        "paddingTop": 32,
      }
    }
    testID="rn-text-input"
    value="Input value"
  />
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#E53930",
        "flexShrink": 1,
        "fontFamily": "Averta CY",
        "fontSize": 14,
        "fontWeight": "normal",
        "letterSpacing": -0.02,
        "lineHeight": 14,
        "marginLeft": 20,
        "marginTop": 12,
      }
    }
  >
    Error message
  </Text>
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#676D79",
        "flexShrink": 1,
        "fontFamily": "Averta CY",
        "fontSize": 13,
        "fontWeight": "normal",
        "left": 20,
        "letterSpacing": -0.02,
        "lineHeight": 13,
        "position": "absolute",
        "top": 12,
      }
    }
  >
    Input label
  </Text>
</View>
`;

exports[`InputText snapshot readOnly 1`] = `
<View>
  <TextInput
    allowFontScaling={false}
    editable={false}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Input label"
    placeholderTextColor="#A4A7AF"
    secureTextEntry={false}
    style={
      {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#E1E2E4",
        "borderRadius": 12,
        "borderWidth": 1,
        "color": "#676D79",
        "fontFamily": "Averta CY",
        "fontSize": 19,
        "fontWeight": "bold",
        "height": 64,
        "letterSpacing": -0.02,
        "lineHeight": 19,
        "paddingBottom": 4,
        "paddingLeft": 20,
        "paddingTop": 32,
      }
    }
    testID="rn-text-input"
    value="Input value"
  />
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#676D79",
        "flexShrink": 1,
        "fontFamily": "Averta CY",
        "fontSize": 14,
        "fontWeight": "normal",
        "letterSpacing": -0.02,
        "lineHeight": 14,
        "marginLeft": 20,
        "marginTop": 12,
      }
    }
  >
    Help message
  </Text>
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#A4A7AF",
        "flexShrink": 1,
        "fontFamily": "Averta CY",
        "fontSize": 13,
        "fontWeight": "normal",
        "left": 20,
        "letterSpacing": -0.02,
        "lineHeight": 13,
        "position": "absolute",
        "top": 12,
      }
    }
  >
    Input label
  </Text>
</View>
`;

exports[`InputText snapshot secureTextEntry 1`] = `
<View>
  <TextInput
    allowFontScaling={false}
    autoCapitalize="none"
    autoComplete="off"
    autoCorrect={false}
    editable={true}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Input label"
    placeholderTextColor="#A4A7AF"
    secureTextEntry={true}
    spellCheck={false}
    style={
      {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#E1E2E4",
        "borderRadius": 12,
        "borderWidth": 1,
        "color": "#676D79",
        "fontFamily": "Averta CY",
        "fontSize": 19,
        "fontWeight": "bold",
        "height": 64,
        "letterSpacing": -0.02,
        "lineHeight": 19,
        "paddingBottom": 4,
        "paddingLeft": 20,
        "paddingRight": 50,
        "paddingTop": 32,
      }
    }
    testID="rn-text-input"
    value="Input value"
  />
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#A4A7AF",
        "flexShrink": 1,
        "fontFamily": "Averta CY",
        "fontSize": 13,
        "fontWeight": "normal",
        "left": 20,
        "letterSpacing": -0.02,
        "lineHeight": 13,
        "position": "absolute",
        "top": 12,
      }
    }
  >
    Input label
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "height": 64,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 64,
      }
    }
    testID="show-or-hidden-password-touchable"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M2.13 12.297a1.097 1.097 0 0 1 0-1.04C4.234 7.37 8.118 4 12 4c3.883 0 7.767 3.37 9.87 7.259.174.323.174.715 0 1.039-2.103 3.888-5.987 7.258-9.87 7.258-3.883 0-7.767-3.37-9.87-7.26Z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4290954697,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="1.8"
        />
        <RNSVGPath
          d="M14.357 9.42a3.333 3.333 0 1 1-4.714 4.715 3.333 3.333 0 0 1 4.714-4.714Z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4290954697,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="1.8"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
